### ISSUE TITLE
[ENHANCEMENT] Performance problems with getHashlist.php

### ISSUE THREAD

**Issue by daymickcorr** [hashtopolis/server#733]
Hashtopolis: 0.12.0 commit 8f5756e branch master 

Hi, i've realised paging is not working correctly and it is affecting performance. 

for example hashtopolis/src/hashlists.php

gets all the results in the database at once and caches it in memory instead of getting first 50 values and on next page get the next 50 in the database

**Comment by TheWorkingDeveloper** [comment]
```
      $(document).ready(function () {
        $('#hashlists').DataTable({
          pageLength: 50,
          "order": [ [0, 'asc'] ],
          "columnDefs": [
            { "orderable": false, "targets": [5, 6] },
            { "orderable": true, "targets": [0, 1, 2, 3, 4] }
          ]
        });
      });
```

That's because it's a jquery function

**Comment by daymickcorr** [comment]
yes paging for the front end works find but it does not work well for the back end, ajax should be used to request a chunk of data, this code works but should be changed 

else {
  //load all hashlists
  $jF = new JoinFilter(Factory::getHashTypeFactory(), HashType::HASH_TYPE_ID, Hashlist::HASH_TYPE_ID);
  $qF1 = new QueryFilter(Hashlist::FORMAT, "" . DHashlistFormat::SUPERHASHLIST, "<>", Factory::getHashlistFactory());
  $qF2 = new ContainFilter(Hashlist::ACCESS_GROUP_ID, Util::arrayOfIds(AccessUtils::getAccessGroupsOfUser(Login::getInstance()->getUser())));
  $joined = Factory::getHashlistFactory()->filter([Factory::JOIN => $jF, Factory::FILTER => [$qF1, $qF2]]);
  /** @var $joinedHashlists Hashlist[] */
  $joinedHashlists = $joined[Factory::getHashlistFactory()->getModelName()];
  $hashlists = array();
  for ($x = 0; $x < sizeof($joinedHashlists); $x++) {
    $hashlists[] = new DataSet(['hashlist' => $joinedHashlists[$x], 'hashtype' => $joined[Factory::getHashTypeFactory()->getModelName()][$x]]);
  }
  UI::add('hashlists', $hashlists);
  UI::add('numHashlists', sizeof($hashlists));
  UI::add('pageTitle', "Hashlists");
}

it would be better if instead of requesting all hashlists when the page is loaded only 50 are asked to the database and cached and then when page 2 is clicked an other 50 queries are request and cached from the database to the db context.

**Comment by daymickcorr** [comment]
it is done well here 
https://github.com/hashtopolis/server/blob/master/src/getHashlist.php
it think this is the backend for the agents 

the backend for the ui should do the same

**Comment by zyronix** [comment]
The enhancement label is more to label it as: not new functions. We are very much aware for this issue.
With the new UI and API it is a little bit improved already. It implements some beter pagination. Maybe you could play around with it? Release 0.14.0 contains a preview of the new UI.

**Comment by ObsidianOracle** [comment]
Closed, since no more input provided from @daymickcorr . Feel free to comment how it behaves today with new pagination etc.
